component/src/main/java/org/wso2/extension/siddhi/io/report/generators/ReportGenerator.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.extension.siddhi.io.report.generators;
+
+import ar.com.fdvs.dj.core.DynamicJasperHelper;
+import ar.com.fdvs.dj.domain.DynamicReport;
+import net.sf.jasperreports.engine.JRDataSource;
+import net.sf.jasperreports.engine.JRException;
+import net.sf.jasperreports.engine.JasperCompileManager;
+import net.sf.jasperreports.engine.JasperExportManager;
+import net.sf.jasperreports.engine.JasperFillManager;
+import net.sf.jasperreports.engine.JasperPrint;
+import net.sf.jasperreports.engine.JasperReport;
+import net.sf.jasperreports.engine.data.JRBeanCollectionDataSource;
+import net.sf.jasperreports.engine.design.JasperDesign;
+import net.sf.jasperreports.engine.export.JRCsvExporter;
+import net.sf.jasperreports.engine.export.ooxml.JRXlsxExporter;
+import net.sf.jasperreports.engine.xml.JRXmlLoader;
+import net.sf.jasperreports.export.SimpleExporterInput;
+import net.sf.jasperreports.export.SimpleOutputStreamExporterOutput;
+import net.sf.jasperreports.export.SimpleWriterExporterOutput;
+import org.wso2.extension.siddhi.io.report.util.DynamicLayoutManager;
+import org.wso2.extension.siddhi.io.report.util.ReportConstants;
+import org.wso2.siddhi.core.exception.SiddhiAppRuntimeException;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * This abstract class provides implementation of methods for the report generation.
+ */
+public abstract class ReportGenerator {
+    private Map<String, String> reportProperties = Collections.EMPTY_MAP;
+
+    public ReportGenerator(Map<String, String> reportProperties) {
+        this.reportProperties = reportProperties;
+    }
+
```

Comment:
Please remove the unnecessary new line

Reply:
Fixed with 736a7f4490bacca5cd67b0b918480e74cc787e12